### PR TITLE
12196 validate usages of :value before saving

### PR DESCRIPTION
**Pull request by cooperka**
Validates for both form "default name" and questioning "default answer".

Example:

Screen Shot 2022-10-31 at 16 48 23

### CODE REVIEW

File: app/validators/forms/dynamic_pattern_validator.rb [hunk 1]
```diff
@@ -12,6 +12,7 @@ def initialize(options)
     def validate(record)
       calc_must_wrap_all_of_default_response_name(record)
       force_calc_for_dollar_refs(record) if force_calc_if&.call(record)
+      refd_option_set_must_exist(record)
```
Comment (jpdimond) [inline app/validators/forms/dynamic_pattern_validator.rb:15]: really easy to understand function name!

File: spec/validators/forms/dynamic_pattern_validator_spec.rb [hunk 2]
```diff
@@ -79,4 +79,61 @@
       end
     end
   end
+
+  describe "option set :value references" do
```
Comment (jpdimond) [inline spec/validators/forms/dynamic_pattern_validator_spec.rb:83]: woo for tests!

**Review by jpdimond — APPROVED** [review]
Really gorgeous code, thanks Kevin!

**Comment by cooperka** [comment]
Specs are flappers!

**Pull request merged**